pkg/minishift/profile/profile.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minishift/config"
+)
+
+//Returns the machine name after taking in to account if --profile or
+//profile information available in allinstance config
+func GetProfileName() string {
+	activeProfile := GetActiveProfile()
+	if constants.ProfileName != constants.DefaultProfileName {
+		return constants.ProfileName
+	} else if activeProfile != "" {
+		return activeProfile
+	} else {
+		return constants.DefaultProfileName
+	}
+}
+
+//Add profile information to allinstancesconfig
+func AddProfileToConfig(name string) error {
+	profile := newProfile(name)
+	config.AllInstancesConfig.Profiles = append(config.AllInstancesConfig.Profiles, profile)
+	err := config.AllInstancesConfig.Write()
+	if err != nil {
+		return fmt.Errorf("Error adding profile information. %s", err)
+	}
+	return nil
+}
+
+func newProfile(name string) config.Profile {
+	return config.Profile{
+		Name:  name,
+		InUse: false,
+	}
+}
+
+//Returns the list of profile names
+func GetProfileNameList() []string {
+	var profileList []string
+	profiles := config.AllInstancesConfig.Profiles
+	for i := range profiles {
+		profile := profiles[i]
+		profileList = append(profileList, profile.Name)
+	}
+	return profileList
+}
+
+//Returns a map of profile name and its InUseinformation
```

Comment:
what does `in use information` mean?

Reply:
It is for checking if a profile is active or not. As I mentioned before I should s/InUse/Active/